### PR TITLE
Added ability to uncraft and change wheels, added in tire armor (#48820)

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
Content "Added ability to uncraft and change wheels using manual tire changer, added in tire armor"

#### Purpose of change
Based on request #48820 - I wanted to add in tire armor but saw this as a good opportunity to add some flexibility to the wheel system and add in the ability to change tires if you find the relevant parts and tool.

#### Describe the solution
 - Changed wheel to use rubber as material
 - Created steel hubs for car wheels, motorbike wheels, wide wheels and bicycle wheels
 - Created relevant tires for wheel, wheel_slick, wheel_wide, wheel_wide_or, wheel_motorbike_wheel_motorbike_or, wheel_bicycle and wheel_bicycle_or - make these nonconductive and all subsequent items
 - Created deflated versions of wheel, wheel_slick, wheel_wide, wheel_wide_or, wheel_motorbike_wheel_motorbike_or, wheel_bicycle and wheel_bicycle_or that are generic items and not wheels
 - Created rubber tire strip and rubber tire chunk resources for crafting and created uncraft recipies for each tire
 - Created manual tire changer furniture for changing tires with ability to also inflate them, added this furniture to garages and tire shops via the mapgen files
 - Created recipes for uncrafting wheel, wheel_slick, wheel_wide, wheel_wide_or, wheel_motorbike_wheel_motorbike_or, wheel_bicycle and wheel_bicycle_or into tires and hubs using manual tire changer (or tool with pry in case of bike tires)
 - Created recipes for creating deflated versions of wheel, wheel_slick, wheel_wide, wheel_wide_or, wheel_motorbike_wheel_motorbike_or, wheel_bicycle and wheel_bicycle_or  using manual tire changer (or tool with pry in case of bike tires)
 - Created recipies for inflating deflated versions of wheel, wheel_slick, wheel_wide, wheel_wide_or, wheel_motorbike_wheel_motorbike_or, wheel_bicycle and wheel_bicycle_or using tools with PRESSURIZATION 1
 - Created Improvised Tire Arm Guards, Improvised Tire Leg Guards and Tireplate as recipes that are available to everyone, made from rubber tire chunks and duct tape (high bash tolerance due to being made from rubber, but very encumbering)
 - Created Dieselpunk Fashion recipe book for housing tailored tire armor recipes
 - Created Tire Legguards, Tire Armguards and Tire Cuirass in regular and XL sizes (based on scrap versions) and added the relevant recipes
 - Created Tire Suit armor based on scrap suit and added the recipies
 - Added all created items to relevant itemgroups so they will spawn as required 

#### Describe alternatives you've considered
It had been suggested that the material for tires may need to be changed to vulcanized rubber, however just using rubber as a material actually makes effective armor that is rightly very encumbering.

#### Testing
All items and recipes have been loaded into a current version of the game and tested both in crafting and uncrafting, the armor has been tested in combat and is very effective but also can overbalance the player if wearing all tire apparel.

### CODE REVIEW

File: data/json/itemgroups/Clothing_Gear/clothing.json [hunk 1]
```diff
@@ -2270,6 +2270,7 @@
       [ "armor_riot", 10 ],
       [ "swat_armor", 6 ],
       [ "armor_scrapsuit", 12 ],
+	  [ "armor_tiresuit", 12 ],
```
Comment (BrettDong) [inline data/json/itemgroups/Clothing_Gear/clothing.json:2273]: You need to format your JSON content. Here's more information on that: https://github.com/CleverRaven/Cataclysm-DDA/wiki/Guide-to-adding-new-content-to-CDDA-for-first-time-contributors#whats-this-about-linting--what-the-heck-is-a-lint-error

Comment (occulti) [reply data/json/itemgroups/Clothing_Gear/clothing.json]: Thanks, I think this should now be resolved, I have run that file through the formatter and pushed it again

Comment (BrettDong) [reply data/json/itemgroups/Clothing_Gear/clothing.json]: Some other files also need to be formatted.

Comment (occulti) [reply data/json/itemgroups/Clothing_Gear/clothing.json]: Thank you, I have gone through all the files now so they all should be done.

File: data/json/itemgroups/Locations_MapExtras/locations.json [hunk 2]
```diff
@@ -1371,6 +1371,26 @@
...
+      [ "tire_motorbike_or", 5 ],
+      [ "tire_slick", 5 ],
+      [ "deflated_wheel_bicycle", 10 ],
+      [ "deflated_wheel", 10 ],
+      [ "deflated_wheel_motorbike", 10 ],
+      [ "deflated_wheel_wide_or", 5 ],
+      [ "deflated_wheel_bicycle_or", 5 ],
+      [ "deflated_wheel_motorbike_or", 5 ],
+      [ "deflated_wheel_slick", 5 ],
+	  [ "hand_pump", 10 ],
```
Comment (Qrox) [inline data/json/itemgroups/Locations_MapExtras/locations.json:1389]:
```suggestion
	    [ "hand_pump", 10 ],
```

File: data/json/items/armor/torso_armor.json [hunk 3]
```diff
@@ -268,6 +316,35 @@
     "material_thickness": 4,
     "material": [ "leather", "bone" ]
   },
+  {
+    "id": "jacket_leather_tire",
+    "repairs_like": "jacket_leather",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": { "str": "tire leather armour" },
```
Comment (Qrox) [inline data/json/items/armor/torso_armor.json:324]:
```suggestion
    "name": { "str_sp": "tire leather armor" },
```
American spelling, and I think the plural is also armor?

**Pull request merged**